File: Python/DecoratorPattern/beverage/condiments/whip.py
from beverage.beverage import Beverage
from beverage.condiments.condiments_decorator import CondimentsDecorator


class Whip(CondimentsDecorator):
    def __init__(self, beverage: Beverage):
        self.beverage = beverage

    def get_description(self) -> str:
        return self.beverage.get_description() + ", Whip"

    def cost(self) -> float:
        return self.beverage.cost() + 0.10

File: DecoratorPattern/Java/Coffee-Hut/src/com/laxmena/beverage/condiments/Whip.java
package com.laxmena.beverage.condiments;

import com.laxmena.beverage.Beverage;

public class Whip extends CondimentDecorator {
    public Whip(Beverage beverage) {
        this.beverage = beverage;
    }
    @Override
    public double cost() {
        return beverage.cost() + 0.10;
    }

    @Override
    public String getDescription() {
        return beverage.getDescription() + ", Whip";
    }
}
